### PR TITLE
fix Silabs SoC families

### PR DESCRIPTION
**Pull request by jerome-pouiller**

### CODE REVIEW

**Comment by jerome-pouiller** [comment]
@nordicjm could you review this PR?

File: soc/silabs/Kconfig.soc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2017 Christian Taedcke
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_VENDOR_SILABS
+	bool
+
+config SOC_FAMILY
+	default "s0" if SOC_FAMILY_SILABS_S0
+	default "s1" if SOC_FAMILY_SILABS_S1
+	default "s2" if SOC_FAMILY_SILABS_S2
```
Comment (nordicjm) [inline soc/silabs/Kconfig.soc:10]:
```suggestion
	default "silabs_s0" if SOC_FAMILY_SILABS_S0
	default "silabs_s1" if SOC_FAMILY_SILABS_S1
	default "silabs_s2" if SOC_FAMILY_SILABS_S2
```

File: drivers/serial/Kconfig.gecko [hunk 2]
```diff
@@ -10,6 +10,6 @@ config UART_GECKO
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
 	select SOC_GECKO_USART
-	select PINCTRL if SOC_GECKO_SERIES1
+	select PINCTRL if SOC_FAMILY_SILABS_S1
```
Comment (nordicjm) [inline drivers/serial/Kconfig.gecko:13]: (unrelated to change) but why is pinctrl only selected for a specific series and not the rest?

File: soc/silabs/s0/efm32hg/Kconfig.soc [hunk 3]
```diff
@@ -13,6 +13,9 @@ config SOC_PART_NUMBER_EFM32HG322F64
 	bool
 	select SOC_SERIES_EFM32HG
 
+config SOC_GECKO_SDID
+	default 77 if SOC_SERIES_EFM32HG
+
```
Comment (nordicjm) [inline soc/silabs/s0/efm32hg/Kconfig.soc:18]: Move all of these to `Kconfig`

**Review by nordicjm — COMMENTED** [review]
Needs to be signed off and follow https://docs.zephyrproject.org/latest/contribute/guidelines.html

File: soc/silabs/soc.yml [hunk 4]
```diff
@@ -1,9 +1,20 @@
 family:
-  - name: silabs_exx32
+  - name: s0
```
Comment (nordicjm) [inline soc/silabs/soc.yml:2]:
```suggestion
  - name: silabs_s0
```

File: soc/silabs/soc.yml [hunk 5]
```diff
@@ -1,9 +1,20 @@
...
     series:
-      - name: efm32gg12b
+      - name: efm32wg
         socs:
-          - name: efm32gg12b810f1024gm64
+          - name: efm32wg990f256
+      - name: efm32hg
+        socs:
+          - name: efm32hg322f64
+  - name: s1
```
Comment (nordicjm) [inline soc/silabs/soc.yml:10]:
```suggestion
  - name: silabs_s1
```

File: soc/silabs/soc.yml [hunk 6]
```diff
@@ -13,40 +24,33 @@ family:
...
+          - name: efr32mg12p433f1024gm68
+      - name: efm32gg11b
         socs:
-          - name: efm32pg1b200f256gm48
+          - name: efm32gg11b820f2048gl192
+          - name: efm32gg11b820f2048gm64
+      - name: efm32gg12b
+        socs:
+          - name: efm32gg12b810f1024gm64
+  - name: s2
```
Comment (nordicjm) [inline soc/silabs/soc.yml:42]:
```suggestion
  - name: silabs_s2
```

File: drivers/serial/Kconfig.gecko (hunk 2 above)
Comment (jerome-pouiller) [reply drivers/serial/Kconfig.gecko:13]: I think `PINCTRL` is also needed for Series 2. For Series 2, the symbol is set in defconfigs of the boards. Obviously, this does not make sense. I believe this symbol should relocate in `soc/silabs/s*/Kconfig.soc`.

**Comment by jerome-pouiller** [comment]
v2:
  - Rename Silabs SoC Families (Jamie)
  - Relocate SDID definitions (Jamie)
  - Fix remaining uses of SOC_GECKO_SERIES*
  - Fix spaces instead of tabs in Kconfig files
  - Fix definitions of SDIDs
  - Fix include paths for Series 2 (sl_device_init_dpll_config.h was not found)
  - Fill commit log bodies
  - Add signed-off (Jamie)

File: soc/silabs/soc.yml [hunk 7]
```diff
@@ -1,9 +1,20 @@
 family:
-  - name: silabs_exx32
```
Comment (nordicjm) [inline soc/silabs/soc.yml:2]: for a follow up (not this PR) could you add `vendor: silabs`

**Review by nordicjm — APPROVED** [review]

**Comment by jerome-pouiller** [comment]
v3:
  - fix occurence of SOC_FAMILY_SILABS_EXX32 in drivers/hwinfo
  - fix spaces in Kconfig

**Review by nordicjm — APPROVED** [review]

**Pull request merged**